url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/search/AlphaBetaSearcher.java

```diff
@@ -0,0 +1,625 @@
+import es.dagaren.gladiator.representation.BitboardSee;
+import es.dagaren.gladiator.representation.Colour;
+import es.dagaren.gladiator.representation.Movement;
+import es.dagaren.gladiator.representation.Position;
+import es.dagaren.gladiator.transposition.Entry;
+import es.dagaren.gladiator.transposition.Table;
+import es.dagaren.gladiator.transposition.Entry.Type;
+
+/**
+ * @author dagaren
+ *
+ */
+public class AlphaBetaSearcher extends Searcher
+{
+   private Logger logger = Logger.getLogger(AlphaBetaSearcher.class);
+   
+   protected long numUpdates;
+   
+   //Constantes de búsqueda
+   protected final int INITIAL_ALPHA = -10000 - 1;
+   protected final int INITIAL_BETA =  10000 + 1;
+   
+   //Arrays de movimientos killer;
+   protected Movement[][] killerMoves = new Movement[this.depthLimit][2];
+   
+   //Comparador MVV LVA
+   protected Comparator<Movement> mvvLvaComparator = new MvvLvaMoveComparator();
+   
+   //Comparador de valor de movimeintos
+   protected Comparator<Movement> moveValueComparator = new MoveValueComparator();
+   
+   //La variante principal
+   protected PrincipalVariation principalVariation = new PrincipalVariation();
+   
+   protected int depth;
+   protected int bestScore;
+   
+   //La tabla de transposicion
+   protected Table transpositionTable = new Table(100000);   //(52428800);
+   
+   protected Movement transpositionMove = null;
+   
+   protected List<Movement> movesForDelete = new ArrayList<Movement>(30);
+   
+   //Variables de estadísticas
+   
+   //Número de cortes producidos
+   protected int cutoffs = 0;
+   //Número de cortes producidos en nodos quiescence
+   protected int qcutoffs = 0;
+   //Número de nodos recorridos en búsqueda normal
+   protected int nodes = 0;
+   //Número de nodos recorridos en búsqueda quiescence
+   protected int qnodes = 0;
+   //
+   protected int iterationNodes = 0;
+   //Número de búsquedas con ventana de aspiración
+   protected int aspirationSearchs = 0;
+   //Número de búsquedas con ventana de aspiración que fallan
+   protected int aspirationSearchsFails = 0;
+   //Número de búsquedas con ventana de aspiración que fallan por inestabilidad
+   protected int aspirationSearchInstabilityFails = 0;
+   //Número de cortes producidos por aciertos en la tabla de transposición
+   protected int transpositionCutoffs = 0;
+   //Número de aciertos con el primer movimiento en la búsqueda
+   protected int firstMoveHits = 0;
+   //Número de aciertos en pvs
+   protected int pvsHits = 0;
+   //Número de fallos de pvs
+   protected int pvsFails = 0;
+   //Búsquedas con movimiento de transposición
+   protected int withTranspositionMove = 0;
+   //Búsquedas sin movimiento de transposición
+   protected int withoutTranspositionMove = 0;
+   //Futility prunings
+   protected int futilityPrunings = 0;
+   
+   
+   
+   int alpha = 0;
+   int beta = 0;
+   
+   int windowSize = 50;
+   
+   @Override
+   public void search()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.